### DISCUSSION TITLE
Trouble updating firmware from 1.0.1 to ... current.

### DISCUSSION THREAD

**Discussion by courtcm2** [ZuluSCSI/ZuluSCSI-firmware#48] [Q&A] [answered]
While the ZuluSCSI works terrific with a couple of my other Macs, I have been struggling on getting the ZuluSCSI working with a Mac Q700 (there is a thread that was opened in one of the user forums, some time after we failed to do it on our own) and it seems to have led at least partially to the implementation of FW update 1.0.6, which I had hoped would lead to a resolution for me.
When I applied the 1.0.7 firmware update over the last weekend, there did not seem to be any change to the behavior that I was seeing, so I started taking a closer look at the files and noticed that the firmware file was still there (it should have been removed during a successful update) and that my firmware on the ZuluSCSI board, as listed in the zululog.txt was still at 1.0.1.  I went back to the website and assumed that I had missed some important information e.g. naming, location, file - something, but it appeared to adhere to the specified rules, so I thought I would try to go to square one.

For my test, I went back to the same Mac (Performa 6115) that I had used for my original (successful) firmware update to 1.0.1, shortly after I bought the ZuluSCSI.
Here is what I did and also what I found:
1) Formatted a new 16GB PNY SD Card as EXFat.
2) Downloaded Zuluscsi firmware ver 1.0.7 from :
    https://github.com/ZuluSCSI/ZuluSCSI-firmware/releases/tag/v1.0.7
3) Since I have a ZuluSCSI V1.1 card, I downloaded “ZuluSCSIv1_1-1.0.7-dc30e55.bin”, using save as from the website.
4) Renamed the file as “ZuluSCSI.bin”
5) Copied the file to the SD Card
(only the firmware file is on the SD card)
6) Loaded the SD Card to the ZuluSCSI board
7) Plugged in the Mac
8) Started the Mac, LED Blinked, but not like it was updating, just like access.
9) Shut down the Mac, removed power, removed the SD card, checked it on a MacBookPro
(see the following zululog.txt that was created and the listing of the files that are on the SD card)

zululog.txt 
[0ms] DIPSW3 is ON: Enabling SCSI termination
[1ms] Optional GreenPAK detected, loading firmware
[53ms] GreenPAK firmware successfully loaded
[166ms] SD card detected, FAT64 volume size: 15093 MB
[166ms] SD MID: 0x74, OID: 0x4A 0x45
[167ms] SD Name: SDU1 
[167ms] SD Date: 12/2012
[167ms] SD Serial: 0x315A3300
[168ms] Config file zuluscsi.ini not found, using defaults
[168ms] Active configuration:
[169ms] -- SelectionDelay: 255
[169ms] -- EnableSCSI2 is on
[186ms] Finding HDD images in directory /:
[187ms] ERROR: No valid images found!
[1688ms] SCSI PHY operating mode: GREENPAK_DMA
[1688ms] Initialization complete!
[1688ms] Platform: ZuluSCSI v1.1
[1689ms] FW Version: 1.0.1 May  6 2022 21:09:11

Works-MacBook-Pro:ZS06 courtcm$ ls -l
total 320
-rwxrwxrwx@ 1 courtcm  staff  101424 Sep  1 18:37 ZuluSCSI.bin
-rwxrwxrwx@ 1 courtcm  staff     703 Dec 31  2021 zululog.txt
Works-MacBook-Pro:ZS06 courtcm$

Previous attempts before this controlled attempt, used other SD cards (including the original, included with the ZuluSCSI), other Macs, both ZuluSCSI cards, so I am hoping that there is something that I missed.  Any suggestions?

**Comment by aperezbios** [comment, marked as answer]
@courtcm2 it's likely you're one of the original ~120 people who got the original AzulSCSI bootloader on their boards, in which case the bootloader is actually searching for a file named AzulSCSI, and not ZuluSCSI* 

If you simply re-name the firmare .bin file, changing ZuluSCSI to AzulSCSI, and that works/applies correctly, then this is the case. You can update to the version of the bootloader that looks for ZuluSCSI* by applying the .bin firmware update using dfu-util. 

The syntax for flashing the .bin file to ZuluSCSI via USB on a Mac (or Linux) is:

`dfu-util -a 0 --dfuse-address 0x08000000 -D firmware.bin`

You'll need to have dfu-util installed on your Mac. It's available through brew, macports, and a number of other sources.

**Comment by courtcm2** [comment]
Thank you for the prompt response, and as you surmised, I did get these immediately after after they were released (before the name change).
I changed the name on the firmware update file, following the rest of the process and was rewarded with a successful firmware update:
[1821ms] Initialization complete!
[1821ms] Platform: ZuluSCSI v1.1
[1821ms] FW Version: 1.0.7-release Aug 25 2022 19:51:14
I will pursue updating the bootloader at a later time (maybe this weekend) to completely alleviate the issue, but I'm thrilled to have an updated ZuluSCSI and also understand why there was an issue with the updates.

Other than this small problem, I really enjoy working with this product and believe it is one of the best and easiest to use SCSI replacement devices out there.
